objects-revisited/main.js

```diff
@@ -0,0 +1,103 @@
+const business = {
+  opens: '9:00 AM',
+  closes: '5:00 PM',
+  totalEmployees: 4,
+  daysOpen: ['M', 'TU', 'W', 'TH', 'F'],
+  employees: {
+    john: {
+      position: 'clerk',
+      daysOfWeekWorking: ['M', 'TU', 'W'],
+    },
+    paul: {
+      position: 'janitor',
+      daysOfWeekWorking: ['W', 'F'],
+    },
+    george: {
+      position: 'reception',
+      daysOfWeekWorking: ['TU', 'W', 'TH', 'F'],
+    },
+    ringo: {
+      position: 'sales',
+      daysOfWeekWorking: ['TU', 'TH'],
+    },
+  },
+};
+
+function addWeekends() {
+  const employees = business.employees;
+  const weekends = ['SA', 'SU'];
+
+  for (let i = 0; i < weekends.length; i++) {
+    business.daysOpen.push(weekends[i]);
+  }
+
+  for (const employee in employees) {
+    for (let i = 0; i < weekends.length; i++) {
+      employees[employee].daysOfWeekWorking.push(weekends[i]);
+    }
+  }
```

Comment:
Works, but we're doing a nested loop here when it's really not necessary. We want to avoid nested loops as much as possible, because it increases the overall time it takes for this function to execute. 

Instead of creating an array for weekends and looping over them, I would just push those values directly. Look at my snippet below.

```
function addWeekends() {
  business.daysOpen.push('S');
  business.daysOpen.push('Su');
  for (const employee in business.employees) {
    const currentEmployee = business.employees[employee];
    currentEmployee.daysOfWeekWorking.push('S');
    currentEmployee.daysOfWeekWorking.push('Su');
  }
}
```

Reply:
Oh yeah that makes sense, that makes it a lot more straightforward.